components/sup/src/main.rs

```diff
@@ -162,6 +179,18 @@ fn cli<'a, 'b>() -> App<'a, 'b> {
             (about: "Start an interactive Bash-like shell")
             (aliases: &["b", "ba", "bas"])
         )
+        (@subcommand apply =>
+            (about: "Applies a configuration to a group of Habitat Supervisors")
+            (@arg SERVICE_GROUP: +required {valid_service_group}
+                "Target service group (ex: redis.default)")
+            (@arg VERSION_NUMBER: +required
+                "A version number (positive integer) for this configuration (ex: 42)")
```

Comment:
This first pass is as close a 1:1 as the previous behaviour. The original reason we chose this was so an external system could keep a version number and pass it here. Since we have no external system for that, we now have made it a habit of just using the date trick.

I'm hesitant to make this auto-generate a version number for you because it could theoretically be destructive. You'll notice that I changed the wording a bit here to apply to be a little less confusing. What we're doing is a PUT or a SET and not an _apply_ which, to me, insinuates we take your previous configuration into account in anyway.

Reply:
Auto-generating the number seems like it might be a nice option to provide at some point. I get the compatibility-with-current-behavior argument, though.

Also, 100% behind removing "apply" wording; that has always bugged me, since it does imply that we take into account what's already there!